src/main/java/io/cdap/plugin/gcp/bigquery/sink/BigQueryMultiSink.java

```diff
@@ -120,6 +122,18 @@ protected void prepareRunInternal(BatchSinkContext context, BigQuery bigQuery, S
     collector.getOrThrowException();
   }
 
+  /**
+   * This method sanitizes outputName when there is an un-allowed special character in dataset
+   * For example: If we have a dataset with outputName testtable$2020, this method will sanitize it to testtable_2020
+   */
+  private String sanitizeOutputName(String outputName) {
+    //Output name before regex: testtable$2020
+    final Pattern compilePattern = Pattern.compile(OUTPUT_PATTERN);
+    final boolean validatePattern = compilePattern.matcher(outputName).matches();
+    //Output name after regex: testtable_2020
+    return (!validatePattern) ? outputName.replaceAll("[^\\p{Alpha}\\p{Digit}]+", "_") : outputName;
```

Comment:
'-' is an allowed character. Seems it will get replaced by '_' here?

Reply:
You're right about '-'. I added in the regex in order to prevent replacement with '_'.